mergify_engine/actions/rebase.py

```diff
@@ -59,20 +60,39 @@ def run(self, ctxt: context.Context, rule: rules.EvaluatedRule) -> check_api.Res
             if output:
                 return output
 
-            try:
-                branch_updater.rebase_with_git(ctxt, self.config["bot_account"])
-                return check_api.Result(
-                    check_api.Conclusion.SUCCESS,
-                    "Branch has been successfully rebased",
-                    "",
-                )
-            except (
-                branch_updater.AuthenticationFailure,
-                branch_updater.BranchUpdateFailure,
-            ) as e:
-                return check_api.Result(
-                    check_api.Conclusion.FAILURE, "Branch rebase failed", str(e)
-                )
+            kind = "REBASE"
+            repo_full_name = ctxt.pull["base"]["repo"]["full_name"]
+            repo_info = ctxt.client.item(f"/repos/{repo_full_name}")
+
+            if repo_info["size"] > config.NOSUB_MAX_REPO_SIZE_KB:
+                if not ctxt.subscription.has_feature(
+                    subscription.Features.LARGE_REPOSITORY
+                ):
+                    ctxt.log.warning(
+                        "repository too big and no subscription active, refusing to %s",
+                        kind,
+                        size=repo_info["size"],
+                    )
+                    raise RuntimeError(
+                        f"{kind} fail: repository is too big and no subscription is active"
+                    )
+                ctxt.log.info("running %s on large repository", kind)
+
+            else:
+                try:
+                    branch_updater.rebase_with_git(ctxt, self.config["bot_account"])
+                    return check_api.Result(
+                        check_api.Conclusion.SUCCESS,
+                        "Branch has been successfully rebased",
+                        "",
+                    )
+                except (
+                    branch_updater.AuthenticationFailure,
+                    branch_updater.BranchUpdateFailure,
+                ) as e:
+                    return check_api.Result(
+                        check_api.Conclusion.FAILURE, "Branch rebase failed", str(e)
+                    )
         else:
```

Comment:
Hmmm, did you try to execute the program in your head first?
That `else` is now triggered by the condition on L67 being `False`, which means that when `repo_info["size"] <= config.NOSUB_MAX_REPO_SIZE_KB` this is going to return "Branch already up to date" without doing anything…